daml/Marketplace/Token.daml

```diff
@@ -58,3 +61,12 @@ template MarketPair
     key id : Id
     maintainer key.signatories
     observer observers
+
+-- |Fetches the 'Token' if the 'AssetType' is a Token, if it is a 'Derivative', fetch the
+-- underlying 'Token'
+tokenOrUnderlying : AssetType -> Id -> Update Token
+tokenOrUnderlying TokenAsset      assetId = snd <$> fetchByKey @Token assetId
+tokenOrUnderlying DerivativeAsset assetId = do
+  (_, deriv) <- fetchByKey @Derivative assetId
+  snd <$> fetchByKey @Token deriv.underlying
+
```

Comment:
I wrote this as a one-liner for funsies but it ended up being way uglier than the `do` block:
`tokenOrUnderlying DerivativeAsset assetId = snd <$> (fetchByKey @Token =<< underlying . snd <$> fetchByKey @Derivative assetId)`